### ISSUE TITLE
Expression Pedal to MIDI CC issue

### ISSUE THREAD

**Issue by BrufordRules** [sigboost-inc/midiglue-forum#4]
Hello,

I have made a simple app to send a MIDI CC from my expression pedal (Roland EV-5) and I have found 2 issues.

- The pedal sends values from 3 to 127 instead of 0 to 127.
- And the worst one. The pedal sends values constantly. Like you were moving the pedal all the time. This cause the DAW to slow down a lot and it's really annoying.

What can I do to stop sending values when the pedal is not moving. And how can I get the whole range of MIDI values.

I have attached a ZIP file with the .flow, and .bin files and a screenshot.

Thank you so much in advance.

Alex

[ExpPedalCC.zip](https://github.com/sigboost-inc/midiglue-forum/files/4451551/ExpPedalCC.zip)

**Comment by kuguma** [comment]
Hello,

Thanks for sharing your questions!

1. You can use the `scale` node and `clip` nodes to map the input to a new range or to limit the range of values.
1. You can use some math nodes to get output only when the value changes significantly. You can also use the `delay_tick` node and `float` node to change the output rate.

Regarding 2, we consider adding a dedicated node to the built-in collection.

---

1. scaleノードとclipノードを使うことで、値を新たな範囲にマップしたり、レンジを制限することができます。
1. 変更のあった場合のみ出力を行いたい場合は、数値比較ノードを組み合わせるか、cyclic_tickノードとfloatノードを組み合わせてリサンプリングを行うことができます。

**Comment by BrufordRules** [comment]
Hi Kai,

First of all, thank you so much for your support.

I have solved the problem of the MIDI range values by extending the R parameter in the _exppedalin_ node to 130 and then subtract 3 to the output value. This way I get 0 at the heel position and 127 at toe position but your solution seems more elegant to me.

About the second issue, I used the _delay_ms_ and _compare_ nodes, to compare the actual value with the delayed one and if they match, close a gate to block the output but it didn't work. I don't know what I did wrong.

Cheers
